distributeaid/aggregated-public-information
# feat(data-ingestion): Upload product items from the needs data to the Strapi collection

## What changed?

This pull request currently populates the `name` field type for the `Product.Category` collection. It relates to task 1 in PR #112. This collection still requires the `items` relation field type to be populated.

Still to do in this PR:

- [x] - Add a `unit` field type to the Strapi `Product.Item` collection.
- [x] - Populate the Strapi `Product.Item` collection, using the needs data to create items with the following fields - `name`, `age_gender`, `size_style`, `category`, `unit`.

**_**Update:_** This pull request now populates the `Product.Category` and `Product.Item` collections.

## How can you test this?
Add the attached [needs-data.json](https://github.com/user-attachments/files/17951499/needs-data.json) file to the `src/scripts/import-needs-assessment-data` folder, setup the `src/scripts/.env` file, then run the dev server in one terminal and `yarn script:import-needs-assessment-data` in another. The logs should show you that 11 categories and 290 products have been created successfully or already exist. (The json file was updated with the full dataset Nov. 28/24)

To test a small amount of data when working with the add-items scripts for the `Product.Item` collection, **use the following demo-data** [needs-data(1).json](https://github.com/user-attachments/files/18495315/needs-data.1.json) instead of the above-noted json file. 
The logs should show you the following breakdown: 
5 categories processed (either in SUCCESS or ALREADY_EXISTS) and 
26 products processed (2 ORIGINAL_DATA_INVALID and 24 in SUCCESS or ALREADY_EXISTS).
